platform/linuxbsd/wayland/display_server_wayland.h

```diff
@@ -0,0 +1,285 @@
+
+		uint32_t flags;
+
+		DisplayServer::WindowMode mode;
+
+		Callable rect_changed_callback;
+		Callable window_event_callback;
+		Callable input_event_callback;
+		Callable drop_files_callback;
+		Callable input_text_callback;
+
+		String title;
+		ObjectID instance_id;
+	};
+
+	struct CustomCursor {
+		RID rid;
+		Point2i hotspot;
+	};
+
+	CursorShape cursor_shape = CURSOR_ARROW;
+	DisplayServer::MouseMode mouse_mode = DisplayServer::MOUSE_MODE_VISIBLE;
+
+	HashMap<CursorShape, CustomCursor> custom_cursors;
+
+	WindowData main_window;
+	WaylandThread wayland_thread;
+
+	Context context;
+
+	bool frame = false;
+	bool emulate_vsync = false;
+
+#ifdef VULKAN_ENABLED
+	VulkanContextWayland *context_vulkan = nullptr;
+	RenderingDeviceVulkan *rendering_device_vulkan = nullptr;
+#endif
+
+#ifdef GLES3_ENABLED
+	EGLManagerWayland *egl_manager = nullptr;
+#endif
+
+#ifdef SPEECHD_ENABLED
+	TTS_Linux *tts = nullptr;
+#endif
+
+#if DBUS_ENABLED
+	FreeDesktopPortalDesktop *portal_desktop = nullptr;
+
+	FreeDesktopScreenSaver *screensaver = nullptr;
+	bool screensaver_inhibited = false;
+#endif
+	static String _get_app_id_from_context(Context context);
+
+	void _send_window_event(WindowEvent p_event);
+
+	static void dispatch_input_events(const Ref<InputEvent> &p_event);
+	void _dispatch_input_event(const Ref<InputEvent> &p_event);
+
+	void _resize_window(Size2i size);
+
+	virtual void _show_window();
+
+public:
+	virtual bool has_feature(Feature p_feature) const override;
+
+	virtual String get_name() const override;
+
+#ifdef SPEECHD_ENABLED
+	virtual bool tts_is_speaking() const override;
+	virtual bool tts_is_paused() const override;
+	virtual TypedArray<Dictionary> tts_get_voices() const override;
+
+	virtual void tts_speak(const String &p_text, const String &p_voice, int p_volume = 50, float p_pitch = 1.f, float p_rate = 1.f, int p_utterance_id = 0, bool p_interrupt = false) override;
+	virtual void tts_pause() override;
+	virtual void tts_resume() override;
+	virtual void tts_stop() override;
+#endif
+
+#ifdef DBUS_ENABLED
+	virtual bool is_dark_mode_supported() const override;
+	virtual bool is_dark_mode() const override;
+#endif
+
+	virtual void mouse_set_mode(MouseMode p_mode) override;
+	virtual MouseMode mouse_get_mode() const override;
+
+	virtual void warp_mouse(const Point2i &p_to) override;
+	virtual Point2i mouse_get_position() const override;
+	virtual BitField<MouseButtonMask> mouse_get_button_state() const override;
+
+	virtual void clipboard_set(const String &p_text) override;
+	virtual String clipboard_get() const override;
+	virtual Ref<Image> clipboard_get_image() const override;
+	virtual void clipboard_set_primary(const String &p_text) override;
+	virtual String clipboard_get_primary() const override;
+
+	virtual int get_screen_count() const override;
+	virtual int get_primary_screen() const override;
+	virtual Point2i screen_get_position(int p_screen = SCREEN_OF_MAIN_WINDOW) const override;
```

Comment:
```suggestion
	virtual Point2i screen_get_position(int p_screen) const override;
```
Default arguments on overrides are unnecessary (and not recommended AFAIK) and clutters here IMO

Reply:
That's very strange, gotcha

Reply:
Looked further into it and not a weirdness just got confused about overrides and calling them